### ISSUE TITLE
Question not really issue.

### ISSUE THREAD

**Issue by ThuGie** [Brian151/OpenShockwave#15]
Hey guys,

So as you know from director itself you can easily load a .dcr file.
Read all the images and sound no problem all with a small piece of code.

So i been doing just that but for some reason all script i read and write to the .dir obtained are empty.
Obvious its because the code itself is empty as well..
aka scriptText

How would i go about creating a .dir with that intact with the bytecode aka only unreadable in the director itself ?

Edit:
Btw .cst files working just fine as well.
I have some .cst i made in the past so no converting needed and getting the code outputted ofcourse not complete but meh good enough.

edit2: 
This
 NumberOfMembers = winRef.movie.member.count
 
turns into this :).

  -- UNK_70 5
  NumberOfMembers = #member.count()

**Comment by Brian151** [comment]
hi!

yeah...

oh, umm...
if you mean STXT sections, those are for something different, but they do have a tendency to be associated with scripting in some form. They're actually "stylized text" , though... One of my recent chnages in docu should reflect this observation...

One would have to completely read the file and re-build it, after creating a decompilation of said bytecode. First, we have to determine if/how a Lscr links back to the cast member where the script's sorce would have been. I'm almost positive this has to happen, because of how a parent script is instantiated...

.cst should work just fine
All that's being done is the linking of LctX, Lnam, and Lscr together. If something more advanced were being done, then there'd be problems, at least for now...

Umm, we need like, your example files, to really be able to do anything... sry

The bottom line is right now, we cannot fully decompile a DXR/DCR/CXT/CCT to DIR/CST. We're missing too much of the required information about the format and such. However, due to the way director compiles things, you can take a dir-opened dir or unprotected cst and re-compile it, assuming the scripts are never edited (;

**Comment by ThuGie** [comment]
Hey thanks for the information.
Would it work out if i created .dir files aka write own code test some stuff,
When i find a unknown just upload both the source + .dcr ?

Also for cst for some reason every file i "unprotect" i end up with empty code upon decompiling.
This is not for all mostly .cct files that i unprotect, for the .dcr's dont have many those that i do fail to unprotect at this time Xtras problem i am assuming.


Edit:
I just tried a .dcr with diropener a file that did work.
And it was able to get quite a bit of code out of it, so its pretty much the files i am struggling with converting that have no code.

But i been testing no way of getting source out of .cct without alreading having the .crt sadly.
Been using code myself to getting .dir aka loading the movie into director and extracting.. wont work getting the code. though diropener manages it. fun though
using diropener getting diropener to dir. then throwing it thru the decompiler gets quite a bit of code. though i dont think enough yet. but its past 12 so i be heading to bed.

**Comment by tomysshadow** [comment]
Unfortunately there script text is lost when going from DIR to DCR which is
ProjectorRays eventual purpose.

**Comment by ThuGie** [comment]
Actually when diropener works, and i convert it to dir i can actually use this tool to get some code out of it, ofcourse not complete depending on what the code is using.

But if i use any other tool, self-made or anything else it wont contain it.
So diropener does some whacky stuff to keep it contained.
but like i said its almost possible to decompile diropener so we can see what it does.

**Comment by Speaqer** [comment]
dirOpener decompresses the dcr and apparently does some byte manipulation after that which I haven't looked into yet.

**Comment by ThuGie** [comment]
If you run the dxr file of diropener thru lscrtoscript you will be able to see the magic,
But the important part i wasnt able to find, i just see it do open(moviefile)
But some parts are still unknown so.. yeah sadly. wish i had the source so i could run it in director itself and see what errors it throws instead of it just closing on me.

**Comment by tomysshadow** [comment]
dirOpener uses Director's SaveMovie function to save the movie. It actually
plays the movie in a window and saves it out from memory. dirOpener itself
doesn't do this task, Director does.

**Comment by ThuGie** [comment]
Problem has been solved for both .cct and .dcr
